presto-tests/src/test/java/com/facebook/presto/memory/TestMemoryManager.java

```diff
@@ -85,72 +81,212 @@ public void shutdown()
         executor = null;
     }
 
-    @Test(timeOut = 240_000)
-    public void testResourceOverCommit()
+    @AfterGroups(groups = {"resourceOverCommit", "clusterPools", "testNoLeak", "resourceOverCommit", "outOfMemoryKiller", "reservedPoolDisabled",
+            "outOfMemoryKillerMultiCoordinator", "reservedPoolDisabledMultiCoordinator", "clusterPoolsMultiCoordinator", "queryUserMemoryLimit"})
+    public void cleanup()
+    {
+        for (TestingPrestoServer server : queryRunner.getServers()) {
+            closeQuietly(server);
+        }
+    }
+
+    @BeforeGroups(groups = {"resourceOverCommit", "clusterPools", "testNoLeak", "queryTotalMemoryLimit"})
+    public void resourceOverCommit()
```

Comment:
maybe  call this simpleQueryRunner() since it's not just resource overcommit anymore.

Reply:
split into different groups as @BeforeGroups and @AfterGroups only executes once per the group set before first method and after last method.

Reply:
Why is that a problem?  I would think that's good since we'll save time by only creating the queryRunner once.

Reply:
The reason is: same `queryRunner` was used by different tests and one of them can close it and causing issue. I split that into 2 sets of `queryRunner` and that helped resolving it.